test/metabase/models/params/chain_filter_test.clj

```diff
@@ -670,3 +671,25 @@
           (testing "`true` if the limit is less than the number of values the field has"
             (is (= true
                    (:has_more_values (chain-filter venues.latitude {venues.price 4} :limit 1))))))))))
+
+(deftest chain-filter-inactive-test
+  (testing "Inactive fields are not used to generate joins"
+    (mt/dataset avian-singles
+      (mt/$ids nil
+        (testing "receiver_id is active and should be used for the join"
+          (memoize/memo-clear! @#'chain-filter/database-fk-relationships)
+          (memoize/memo-clear! @#'chain-filter/find-joins)
+          (is (= [{:lhs {:table $$messages, :field %messages.receiver_id}
+                   :rhs {:table $$users, :field %users.id}}]
+                 (#'chain-filter/find-joins (mt/id) $$messages $$users))))
+
+        (try
+          (t2/update! :model/Field {:id %messages.receiver_id} {:active false})
+          (testing "check that it switches to sender once receiver is inactive"
+            (memoize/memo-clear! @#'chain-filter/database-fk-relationships)
+            (memoize/memo-clear! @#'chain-filter/find-joins)
+            (is (= [{:lhs {:table $$messages, :field %messages.sender_id}
+                     :rhs {:table $$users, :field %users.id}}]
+                   (#'chain-filter/find-joins (mt/id) $$messages $$users))))
+          (finally
+            (t2/update! :model/Field {:id %messages.receiver_id} {:active true})))))))
```

Comment:
That doesn't look fantastic, but `(t2/with-transaction [... {:rollback-only true}] ...)` did not work for me to stop the changes unfortunately. Anyway, clearing memoization cache doesn't look like it could be done in parallel either.